tools/generate_code_cache.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const natives = process.binding('natives');
+const vm = require('vm');
+const fs = require('fs');
+let count = 0;
+
+function human(num) {
+  if (num < 1024) {
+    return `${(num).toFixed(2)}B`;
+  } else if (num < 1024 ** 2) {
+    return `${(num / 1024).toFixed(2)}KB`;
+  } else if (num < 1024 ** 3) {
+    return `${(num / (1024 ** 2)).toFixed(2)}MB`;
+  } else {
+    return `${(num / (1024 ** 3)).toFixed(2)}GB`;
+  }
+}
+
+/**
+ * @param {string} key
+ * @param {Buffer} cache
+ */
+function getInitalizer(key, cache) {
+  const defName = key.replace(/\//g, '_').replace(/-/g, '_');
+  const def = `static uint8_t ${defName}_raw[] = {\n` +
+                     `${cache.join(',')}\n};`;
+  const initializer = `
+  v8::Local<v8::ArrayBuffer> ${defName}_ab =
+    v8::ArrayBuffer::New(isolate, ${defName}_raw, ${cache.length});
+  v8::Local<v8::Uint8Array> ${defName}_array =
+    v8::Uint8Array::New(${defName}_ab, 0, ${cache.length});
+  target->Set(context,
+              OneByteString(isolate, "${key}"),
+              ${defName}_array).FromJust();
+  `;
+  return {
+    def, initializer
+  };
+}
+
+const cacheDefs = [];
+const initializers = [];
+
+for (const key of Object.keys(natives)) {
+  if (key === 'config') continue;
+  const wrapper = [
```

Comment:
Currently there are no way to get the native module wrapper in the user land so this is just copy-pasting. We could put the wrapper in a file and produce this using macros in js2c and read it off the disk here, or just put a comment here. In any cases if the source do not match, the module would appear in `process.cacheRejectedList`

Reply:
require('module).wrapper?

Reply:
@benjamingr That's the wrapper we use for user land modules. The native modules uses `NativeModule.wrapper` which is internal and subject to change.